### PR TITLE
page-object-model

### PR DESCRIPTION
**Pull request by dgananp**

### CODE REVIEW

File: src/page/AddressStep.page.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class AddressStepPage {
+ private get proccedToCheckout(): ElementFinder {
```
Comment (aperdomob) [inline src/page/AddressStep.page.ts]: add Button suffix

File: src/page/BankPayment.page.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class BankPaymentPage {
+ private get confirmOrder(): ElementFinder {
```
Comment (aperdomob) [inline src/page/BankPayment.page.ts]: add Button suffix

File: src/page/OrderResume.page.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class OrderResumePage {
+ private get orderConfirmation(): ElementFinder {
```
Comment (aperdomob) [inline src/page/OrderResume.page.ts]: add Label suffix

File: src/page/PaymentStep.page.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class PaymentStepPage {
+ private get payByBankWire(): ElementFinder {
```
Comment (aperdomob) [inline src/page/PaymentStep.page.ts]: add Button suffix

File: src/page/ProductAddedModal.page.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class ProductAddedModalPage {
+ private get proccedToCheckout(): ElementFinder {
```
Comment (aperdomob) [inline src/page/ProductAddedModal.page.ts]: add Button suffix

File: src/page/ShippingStep.page.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class ShippingStepPage {
+
+ private get termOfService(): ElementFinder {
```
Comment (aperdomob) [inline src/page/ShippingStep.page.ts]: add Check suffix

File: src/page/ShippingStep.page.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+   return $('#cgv');
+ }
+
+ private get proccedToCheckout(): ElementFinder {
```
Comment (aperdomob) [inline src/page/ShippingStep.page.ts]: add Button suffix

File: src/page/ShippingStep.page.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  return $('#form > p > button > span');
+ }
+
+ public checkTermOfService(): promise.Promise<void> {
```
Comment (aperdomob) [inline src/page/ShippingStep.page.ts]: make an only method with these two steps

File: src/page/ShippingStep.page.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+   return this.termOfService.click(); 
+ }
+
+ public goToPaymentStep(): promise.Promise<void> {
+    return this.proccedToCheckout.click(); 
+  }
+}
```
Comment (aperdomob) [inline src/page/ShippingStep.page.ts]: EOF

File: src/page/SignInStep.page.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+ private get passwd(): ElementFinder {
+  return $('#passwd');
+ }
+
+ private get submitLogin(): ElementFinder {
+  return $('#SubmitLogin > span');
+ }
+
+ public signIn(email, passwd): promise.Promise<void> {
```
Comment (aperdomob) [inline src/page/SignInStep.page.ts]: use async/await in this method

File: src/page/SummaryStep.page.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class SummaryStepPage {
+ private get proccedToCheckout(): ElementFinder {
```
Comment (aperdomob) [inline src/page/SummaryStep.page.ts]: add Button suffix

File: src/page/SignInStep.page.ts [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+   this.email.sendKeys(email);
+   this.passwd.sendKeys(passwd);
+   return this.submitLogin.click(); 
+ }
+}
```
Comment (aperdomob) [inline src/page/SignInStep.page.ts]: EOF

File: test/BuyTshirt.spec.ts [hunk 13]
```diff
@@ -1,44 +1,64 @@
-import { $, browser } from 'protractor';
+import { browser } from 'protractor';
+import { MenuContentPage } from '../src/page';
```
Comment (aperdomob) [inline test/BuyTshirt.spec.ts]: import all classes in the same import

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: src/page/AddressStep.page.ts (hunk 1 above)
Comment (dgananp) [reply src/page/AddressStep.page.ts]: Do i need to change the name proccedToCheckout --> addButtonSuffix() ??
or What do you want say with add Button suffix?

Comment (dgananp) [reply src/page/AddressStep.page.ts]: Don`t worry, i could understand, Just, I knew the word suffix.

File: src/page/ProductDetail.page.ts [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class ProductDetailPage {
+ private get addToCard(): ElementFinder {
```
Comment (aperdomob) [inline src/page/ProductDetail.page.ts]: Button suffix

File: src/page/ProductList.page.ts [hunk 15]
```diff
@@ -0,0 +1,11 @@
+import { $, ElementFinder, promise } from 'protractor';
+
+export class ProductListPage {
+ private get chooseProduct(): ElementFinder {
```
Comment (aperdomob) [inline src/page/ProductList.page.ts]: Button suffix

File: src/page/ShippingStep.page.ts [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+
+ private get termOfServiceCheck(): ElementFinder {
+   return $('#cgv');
+ }
+
+ private get proccedToCheckoutButton(): ElementFinder {
+  return $('#form > p > button > span');
+ }
+ 
+ public goToPaymentStep(): promise.Promise<void> {
```
Comment (aperdomob) [inline src/page/ShippingStep.page.ts]: usea async/await

File: src/page/SignInStep.page.ts [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+
+ private get passwd(): ElementFinder {
+  return $('#passwd');
+ }
+
+ private get submitLogin(): ElementFinder {
+  return $('#SubmitLogin > span');
+ }
+
+ async  signIn(email, passwd) {
```
Comment (aperdomob) [inline src/page/SignInStep.page.ts]: public async signIn(email, passwd): Promise<void>

File: test/BuyTshirt.spec.ts [hunk 18]
```diff
@@ -1,44 +1,54 @@
-import { $, browser } from 'protractor';
+import { browser } from 'protractor';
+import { MenuContentPage, SignInStepPage, ProductListPage, ProductDetailPage,
```
Comment (aperdomob) [inline test/BuyTshirt.spec.ts]:
```
import {
  MenuContentPage,
  SignInStepPage,
  ProductListPage,
  ProductDetailPage,
  ...,
  OrderResumePage
} from '../src/page';
```

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: src/page/SignInStep.page.ts [hunk 19]
```diff
@@ -14,7 +14,7 @@ export class SignInStepPage {
   return $('#SubmitLogin > span');
  }
 
- async  signIn(email, passwd) {
+ public async  signIn(email, passwd): Promise<void> {
```
Comment (aperdomob) [inline src/page/SignInStep.page.ts]: hay dos espacios entre el async y el signIn

**Review by aperdomob — CHANGES_REQUESTED** [review]

File: src/page/SignInStep.page.ts (hunk 19 above)
Comment (aperdomob) [reply src/page/SignInStep.page.ts]: Falta esta comentario

**Review by aperdomob — APPROVED** [review]

**Pull request merged**